components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/authenticator/uaepass/UAEPassAuthenticator.java

```diff
@@ -340,24 +398,32 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
                     forEach(entry -> buildClaimMappings(claims, entry, attributeSeparator));
 
             if (StringUtils.isBlank(authenticatedUserId)) {
-                throw new AuthenticationFailedException("Cannot find the userId from the id_token sent "
-                        + "by the federated IDP.");
+                setAuthenticatorMessageToContext(UAEPassAuthenticatorConstants.ErrorMessages.
+                        USER_ID_NOT_FOUND_IN_ID_TOKEN, context);
+                throw new AuthenticationFailedException(UAEPassAuthenticatorConstants.ErrorMessages.
+                        USER_ID_NOT_FOUND_IN_ID_TOKEN.getMessage());
             }
             authenticatedUser = AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier
                     (authenticatedUserId);
             authenticatedUser.setUserAttributes(claims);
             context.setSubject(authenticatedUser);
 
         } catch (UAEPassAuthnFailedException e) {
+            setAuthenticatorMessageToContext(UAEPassAuthenticatorConstants.ErrorMessages.
```

Comment:
Why are we setting the error message to the context?

Reply:
This is read from the app native auth code to send the error details in the api response.